app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -11,6 +11,10 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
     def handle(provider)
       auth_hash = request.env['omniauth.auth']
+      unless auth_hash.info.email
+        flash[:error] = "Email field is missing in your #{provider} account"
+        return redirect_to new_user_registration_path
```

Comment:
The error you encountered tells you exactly what you need to do to fix it. 
`redirect_to new_user_registration_path and return`

Reply:
Yes, this is exactly what I thought. So, I guess this is already correct?Do I need to make any changes in it?

Reply:
Read my comment :)